hirundo/dataset_optimization.py

```diff
@@ -296,32 +296,45 @@ def check_run_by_id(
         """
         logger.debug("Checking run with ID: %s", run_id)
         with logging_redirect_tqdm():
-            t = tqdm(total=100)
+            t = tqdm(total=100.0)
             for iteration in self._check_run_by_id(run_id):
                 if iteration["state"] == "SUCCESS":
                     t.set_description("Optimization run completed successfully")
-                    t.update(100)
+                    t.n = 100.0
+                    t.refresh()
                     t.close()
                     return iteration["result"]
                 elif iteration["state"] == "PENDING":
                     t.set_description("Optimization run queued and not yet started")
-                    t.update(0)
+                    t.n = 0.0
+                    t.refresh()
                 elif iteration["state"] == "STARTED":
-                    t.set_description("Optimization run in progress")
-                    t.update(0)
+                    t.set_description(
+                        "Optimization run in progress. Downloading dataset"
+                    )
+                    t.n = 0.0
+                    t.refresh()
                 elif iteration["state"] is None:
                     if (
                         iteration["result"]
                         and iteration["result"]["result"]
                         and isinstance(iteration["result"]["result"], str)
                     ):
-                        t.update(
-                            int(iteration["result"]["result"].removesuffix("% done"))
+                        current_progress_percentage = float(
+                            iteration["result"]["result"].removesuffix("% done")
+                        )
+                        desc = (
+                            "Optimization run completed. Uploading results"
+                            if current_progress_percentage == 100.0
+                            else "Optimization run in progress"
                         )
-                        logger.debug("Checking run with ID: %s", run_id)
+                        t.set_description(desc)
+                        t.n = current_progress_percentage
+                        t.refresh()
```

Comment:
can you do t.refresh() outside each case, on every iteration? it seems to happen for each case on its own

Reply:
The issue would be that renders before closing the progress bar would not be rendered, e.g.
<img width="487" alt="image" src="https://github.com/user-attachments/assets/643eabe2-0eb0-4c4f-8a86-3c3f3a59a0b8">
OR
<img width="580" alt="image" src="https://github.com/user-attachments/assets/61b0862e-1672-451f-9f6a-87580d48ecdc">
In theory, this should happen first:
<img width="635" alt="image" src="https://github.com/user-attachments/assets/0d30884f-2c20-4cae-bd81-1b63b60a5f75">
But in practice, messages can be lost, so I don't know if it is safe to only render after potentially closing the progress bar. What do you think @tomerr-hirundo ?

Reply:
ok, so leave it as it is